.github/workflows/bootstrap/action.yml

```diff
@@ -0,0 +1,39 @@
+---
+name: Bootstrap Checkout
+description: Ensures all actions bootstrap the same
+
+outputs:
+  full-version:
+    description: "The current full version number"
+    value: ${{ steps.dotnet.outputs.full-version }}
+  major-version:
+    description: "The current major version number, semver"
+    value: ${{ steps.dotnet.outputs.major-version }}
+    
+runs:
+  using: "composite"
+  steps:
+    # Ensure we fetch all tags
+    - shell: bash
+      run: |
+        git fetch --prune --unshallow --tags
+        git tag --list
```

Comment:
https://github.com/actions/checkout has the `fetch-depth` input. If you use a value of `0`, it will fetch all tags.

But ignore me, if this is some special case.

Reply:
I believe this is the most lightweight method to query the repositories tags. 

We use those to determine the version number.